### ISSUE TITLE
NVDA does not announce disabled state of disabled buttons (Java Access Bridge API)

### ISSUE THREAD

**Issue by anOolamr** [nvaccess/nvda#13898]
NVDA does not announce the disabled state of a disabled button. This affects all class instances which are derived from AbstractButton: JButton, JToggleButton, JRadioButton, JCheckBox, JMenuitem, JRadioButtonMenuItem, JCheckBoxMenuItem. 

Some background information: For a disabled button instance, the enabled property is set to false. The effect is that the AccessibleStateSet of the related AccessibleObject does not contain the AccessibleState enabled. Unfortunately Java Swing does not allow that a disabled component is focusable and focustraversable (to be announced by screen readers). For Accessibility reasons, developers may keep the instance enabled (and hence focusable and focustraversable). In this case they make the button instance displaying as "disabled", disable the actions and override the getAccessibleStateSet to remove the enabled property.

I wrote a simple application Test_Disabled_Button_Announcement.jar and attached it to this bug report. (If you prefer a .java file, you can find it within the .jar file.) 
[Test_Disabled_Button_Announcement.zip](https://github.com/nvaccess/nvda/files/9103770/Test_Disabled_Button_Announcement.zip)

The jar-file opens a frame containing 
- a group of two checkboxes, thereof one disabled checkbox
- two buttons, thereof one disabled button
- a group of three radio buttons, thereof one disabled radio button

Press command NVDA-key + B to read all UI components.
Disabled UI components should be announced additionally as "not available".

### Steps to reproduce:
1. Extract the Test_Disabled_Button_Announcement.jar file from the linked .zip file.

2. Execute the attached .jar file with an up-to-date JRE version.
-> A Java window with title "Frame with disabled buttons" opens.
-> The "First checkbox" has the keyboard focus.
-> NVDA announces: "Frame with disabled buttons, Checkbox group, panel, First checkbox, checkbox, not checked".

3. Press NVDA-key + B to announce all UI components.
-> NVDA announces the frame title and all UI components of the application.

### Actual behavior:
NVDA does announces the disabled UI components as disabled. These UI components appear to the screen reader users as enabled.

### Expected behavior:
For disabled UI components, NVDA should announce the disabled state explicitly, e.g. with state information "not applicable".

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
Version: 2022.1

#### Windows version:
Version: 10.0.19043 Build 19043

#### Name and version of other software in use when reproducing the issue:
java version "18.0.1" 2022-04-19
Java(TM) SE Runtime Environment (build 18.0.1+10-24)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.1+10-24, mixed mode, sharing)

#### Other information about your system:
Java Access Bridge is enabled.

#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, same behavior.

#### If NVDA add-ons are disabled, is your problem still occurring?
Not applicable.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not applicable.

**Comment by mwhapples** [comment]
This sounds like an issue I have already fixed which is pending for a future release. See the pullrequest #13711 and test if that fixes it or try running NVDA from a build of master.

**Comment by seanbudd** [comment]
This should be fixed in 2022.2rc1

**Comment by anOolamr** [comment]
Retested with version 2022.2rc1 by running nvda_2022.2rc1.exe:
Missing state "enabled" in AccessibleStateSet is announced as "unavailable", which means that issue is fixed.

Please include translation for "unavailable", as "unavailable" is also announced for this state when NVDA version 2022.2rc1 runs in language German.

Using keystroke NVDA+B in attached application Test_Disabled_Button_Announcement.jar does not include disabled elements in reading all any more. Is this intended or a side effect?

**Comment by anOolamr** [comment]
@mwhapples No, unchecking this option does not read out the unavailable controls.
And in installed NVDA 2022.1 version this option is checked and unavailable controls are read out.

**Comment by mwhapples** [comment]
Do unavailable controls get spoken in a native windows application? I.E. is this specific to Java swing? If unavailable controls are not spoken in native windows applications then it probably is intended.
